podcasts/ShelfActionsViewController+Table.swift

```diff
@@ -51,11 +51,19 @@ extension ShelfActionsViewController: UITableViewDelegate, UITableViewDataSource
             } else {
                 cell.actionIcon.tintColor = ThemeColor.playerContrast02()
             }
+
+            // Disable transcript if not available
+            let isTranscriptsAndIsDisable = action == .transcript && !isTranscriptEnabled
```

Comment:
Not particularly proud of this. My first try was using a property on the `enum`, but with the async nature of checking transcript availability I couldn't make it work.

I extracted the logic to check for transcripts to a protocol, so there's no code duplication.

Given we will eventually remove the button from the shelf I think this is a ok trade-off for now.